google/utils.go

```diff
@@ -124,73 +124,50 @@ func getZonalBetaResourceFromRegion(getResource func(string) (interface{}, error
 	return nil, nil
 }
 
-// getNetworkLink reads the "network" field from the given resource data and if the value:
+// getNetworkLink takes a "network" field and if the value:
 // - is a resource URL, returns the string unchanged
 // - is the network name only, then looks up the resource URL using the google client
-func getNetworkLink(d *schema.ResourceData, config *Config, field string) (string, error) {
-	if v, ok := d.GetOk(field); ok {
-		network := v.(string)
-
-		project, err := getProject(d, config)
-		if err != nil {
-			return "", err
-		}
-
-		if !strings.HasPrefix(network, "https://www.googleapis.com/compute/") {
-			// Network value provided is just the name, lookup the network SelfLink
-			networkData, err := config.clientCompute.Networks.Get(
-				project, network).Do()
-			if err != nil {
-				return "", fmt.Errorf("Error reading network: %s", err)
-			}
-			network = networkData.SelfLink
-		}
+func getNetworkLink(config *Config, project, network string) (string, error) {
```

Comment:
In #527 and #529 I created a helper function to replace this method. I started the work of moving off the `getNetworkLink` method, my next step was to finish off with instance and instance template. Consider deleting this method and use the method [here](https://github.com/terraform-providers/terraform-provider-google/blob/master/google/field_helpers.go#L20). The problem with this method is that it makes an extra network call to generate the self link for the network. 

The other benefit of using this new method is that you can easily extract the project from the link. No need for the extra methods you added in utils.go to do this.

Reply:
I replaced `getNetworkLink` with the helpers you suggested, thanks!

I added some code to parse regional fields, with the intention of using the same helpers for subnetworks. Unfortunately in the context of an instance/instance template the subnetwork's project isn't necessarily the same as the instance's project due to XPN, so I don't think the existing helpers will work.